frontend/integration-tests/tests/environment.scenario.ts

```diff
@@ -0,0 +1,113 @@
+import { browser, ExpectedConditions as until } from 'protractor';
+import { safeLoad, safeDump } from 'js-yaml';
+import * as _ from 'lodash';
+
+import { appHost, testName, checkLogs, checkErrors } from '../protractor.conf';
+import * as crudView from '../views/crud.view';
+import * as environmentView from '../views/environment.view.ts';
+import * as yamlView from '../views/yaml.view';
+
+browser.ignoreSynchronization = true;
+
+const BROWSER_TIMEOUT = 15000;
+const WORKLOAD_NAME = `env-${testName}`;
+const Actions = {
+  add: 'add',
+  delete: 'delete',
+};
+
+describe('Environment Page', () => {
+
+  beforeAll(async() => {
+    await browser.get(`${appHost}/k8s/ns/${testName}/daemonsets`);
+    await crudView.isLoaded();
+    await crudView.createYAMLButton.click();
+    await yamlView.isLoaded();
+    const content = await yamlView.editorContent.getText();
+    const newContent = _.defaultsDeep({}, {metadata: {name: WORKLOAD_NAME, labels: {['lbl-env']: testName}}}, safeLoad(content));
+    await yamlView.setContent(safeDump(newContent));
+    await crudView.saveChangesBtn.click();
+    checkLogs();
+    checkErrors();
+  });
+
+  afterEach(() => {
+    checkLogs();
+    checkErrors();
+  });
+
+  afterAll(async() => {
+    await browser.get(`${appHost}/k8s/ns/${testName}/daemonsets`);
+    await crudView.isLoaded();
+    await crudView.nameFilter.sendKeys(WORKLOAD_NAME);
+    await browser.wait(until.elementToBeClickable(crudView.resourceRowNamesAndNs.first()), BROWSER_TIMEOUT);
+    await crudView.deleteRow('daemonset')(WORKLOAD_NAME);
+    checkLogs();
+    checkErrors();
+  });
+
+  const validateKeyAndValue = async function ( key: string,
+    value: string,
+    isPresent: boolean
+  ) {
+    let keyFound = 0;
+
+    const envKey = await environmentView.rowsKey.getAttribute('value');
+    const envValue = await environmentView.rowsValue.getAttribute('value');
+
+    if (envKey === key){
+      keyFound = keyFound + 1;
+      expect( envValue ).toBe(value);
+    }
+
+    if (isPresent){
+      expect(keyFound).toEqual(1);
+    } else {
+      expect(keyFound).toEqual(0);
+    }
+  };
+
+  const environmentEditor = async function (
+    action: string,
+    key: string,
+    value: string
+  ) {
+    await browser.get(`${appHost}/k8s/ns/${testName}/daemonsets/${WORKLOAD_NAME}/environment`);
+
+    switch (action) {
+      case Actions.add: {
+        await environmentView.addVariable(key, value);
+        break;
+      }
+      case Actions.delete: {
+        await environmentView.deleteVariable();
+        break;
+      }
+      default: {
+        throw new Error(`Invalid action [${action}]`);
+      }
+    }
+
+    await environmentView.isLoaded();
+    await crudView.isLoaded();
+  };
+
```

Comment:
This comment is helpful.  Can you do this with `describe()` wrappers?

Reply:
I think we would want this as output if the test ever failed.

Reply:
(And I have no issue with multiple levels of `describe()` personally).

Reply:
Added in new commit; please take a look!

Reply:
Everything above looks great!  Curious if we can tinker with this one a bit more, if we can encode these comments into `describe()` wrappers:

```JavaScript 
+  // Scenario:
+  // Given I log in into the console if it's required
+  // And I create a daemonset
+  // And I open the Environment page
+  // And I add an environment variable
+  // When I delete the variable
+  // Then I expect that the variable is not displayed
```

Reply:
```JavaScript 
+  // Given I log in into the console if it's required
+  // And I create a daemonset
```
@benjaminapetersen I didn't write these parts of the scenario; they're from the other crud tests. It looks like there is already separate testing for using YAML to create daemonsets, etc. in the main crud test. Do you want that duplicated?

I can add a test to make sure the Environment Page tab is selected if that's something you'd find valuable - let me know.

Reply:
Ah, I see what you are saying.  Interesting.

@alecmerdler any chance you know why these tests have the big comment blocks?  Is this an artifact of someone coming from a Cucumber background, or is it something we want to continue going forward?

Reply:
They were written by an external contractor and really should be removed because it's redundant with Jasmine's BDD syntax.

Reply:
Alright, I removed it.